src/App.tsx

```diff
@@ -15,71 +17,80 @@ export const goods = [
+  render() {
+    const { selectedGood } = this.state;
 
-      {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-      <button
-        data-cy="ClearButton"
-        type="button"
-        className="delete ml-3"
-      />
-    </h1>
+    const setRightState = (good: string) => {
+      this.setState({
+        selectedGood: selectedGood === good
+          ? ''
+          : good,
+      });
+    };
 
-    <table className="table">
-      <tbody>
-        <tr data-cy="Good">
-          <td>
-            <button
-              data-cy="AddButton"
-              type="button"
-              className="button"
-            >
-              +
-            </button>
-          </td>
-
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Dumplings
-          </td>
-        </tr>
+    return (
+      <main className="section container">
+        <h1 className="title is-flex is-align-items-center">
+          {selectedGood
+            ? `${selectedGood} is selected`
+            : 'No goods selected'}
 
-        <tr data-cy="Good" className="has-background-success-light">
-          <td>
+          {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
+          {selectedGood
+          && (
+            // eslint-disable-next-line jsx-a11y/control-has-associated-label
             <button
-              data-cy="RemoveButton"
+              data-cy="ClearButton"
               type="button"
-              className="button is-info"
-            >
-              -
-            </button>
-          </td>
+              className="delete ml-3"
+              onClick={() => {
+                this.setState({ selectedGood: '' });
+              }}
+            />
+          )}
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Jam
-          </td>
-        </tr>
+        </h1>
 
-        <tr data-cy="Good">
-          <td>
-            <button
-              data-cy="AddButton"
-              type="button"
-              className="button"
-            >
-              +
-            </button>
-          </td>
+        <table className="table">
+          <tbody>
+            {goods.map((good: string) => (
+              <tr
+                key={good}
+                data-cy="Good"
+                className={classNames(
+                  { 'has-background-success-light': selectedGood === good },
+                )}
+              >
+                <td>
+                  <button
+                    className={selectedGood === good
+                      ? 'button is-info'
+                      : 'button'}
+                    data-cy={selectedGood === good
+                      ? 'RemoveButton'
+                      : 'AddButton'}
+                    type="button"
+                    onClick={() => setRightState(good)}
+                  >
+                    {selectedGood === good
+                      ? '-'
+                      : '+'}
```

Comment:
later you can decompose button into separate component and pass prop `isGoodSelected` - and prepare propper `data-cy` and text before render and use constants-strings